src/mantine-core/src/Rating/Rating.tsx

```diff
@@ -0,0 +1,302 @@
+  /** Rating default value */
+  defaultValue?: number;
+
+  /** Rating value */
+  value?: number;
+
+  /** This icon will be displayed if empty */
+  emptySymbol?: React.ReactNode;
+
+  /** This icon will be displayed if full */
+  fullSymbol?: React.ReactNode;
+
+  /** Rating Fraction, by default it is 1 */
+  fractions?: number;
+
+  /** Size */
+  size?: MantineSize;
+
+  /** No of ratings */
+  count?: number;
+
+  /** called on onChange event */
+  onChange?: (value: number) => void;
+
+  /** called if hover change */
+  onChangeHover?: (value: number) => void;
+
+  /** return labelText for a symbol, by default the function return the value */
+  getSymbolLabel?: (value: number) => string;
+
+  /** name of rating, should be unique within the page */
+  name?: string;
+
+  /** Readonly */
+  readonly?: boolean;
+}
+
+const defaultProps: Partial<RatingProps> = {
+  size: 'sm',
+  getSymbolLabel: (value) => `${value}`,
+  count: 5,
+  fractions: 1,
+  readonly: false,
+};
+
+export const Rating = forwardRef<HTMLInputElement, RatingProps>((props, ref) => {
+  const {
+    defaultValue,
+    value,
+    emptySymbol,
+    fullSymbol,
+    size,
+    count,
+    fractions,
+    onChange,
+    onChangeHover,
+    getSymbolLabel,
+    name,
+    readonly,
+    className,
+    classNames,
+    styles,
+    unstyled,
+    onMouseEnter,
+    onMouseMove,
+    onMouseLeave,
+    ...others
+  } = useComponentDefaultProps('Rating', defaultProps, props);
+
+  const { classes, cx, theme } = useStyles(
+    { size, readonly },
+    { name: 'Rating', classNames, styles, unstyled }
+  );
+
+  const _name = useId();
+  const rootRef = useRef<HTMLDivElement>(null);
+  const [stableValue, setStableValue] = useUncontrolled({
+    value,
+    defaultValue,
+    finalValue: 0,
+  });
+  const decimalUnit = 1 / fractions;
+
+  const [interactiveValue, setInteractiveValue] = useState(-1);
+  const [isClientOutside, setClientOutside] = useState(true);
+
+  const stableValueRounded = roundValueTo(stableValue, decimalUnit);
+  let _value = stableValueRounded;
+  if (interactiveValue !== -1) _value = interactiveValue;
+
+  const handleMouseEnter = (event: React.MouseEvent<HTMLDivElement>) => {
+    if (onMouseEnter) onMouseEnter(event);
+    setClientOutside(false);
+  };
+
+  const handleMouseMove = (event: React.MouseEvent<HTMLDivElement>) => {
+    if (onMouseMove) onMouseMove(event);
+    const { left, right, width } = rootRef.current.getBoundingClientRect();
+
+    const symbolWidth = width / count;
+
+    let hoverPosition;
+    if (theme.dir === 'rtl') hoverPosition = right - event.clientX;
+    else hoverPosition = event.clientX - left;
+
+    const hoverValue = hoverPosition / symbolWidth;
+
+    let rounded = roundValueTo(hoverValue + decimalUnit / 2, decimalUnit);
+
+    rounded = clamp(rounded, decimalUnit, count);
+
+    setInteractiveValue(rounded);
+    if (onChangeHover && rounded !== interactiveValue) onChangeHover(rounded);
```

Comment:
Do not use single line ifs, use it only with brackets `if () {}`, like on line 181